File: Go/ExampleTests/common/result_string.go
package common

type ResultString struct {
	AddressMatches string
}

func NewResultStringFromSlice(v []string) ResultString {
	s := ResultString{}
	if len(v) > 0 { s.AddressMatches = v[0] }
	return s
}

// NewResultStringFromText builds from the text form, e.g. Money as "25 USD".
func NewResultStringFromText(text string) ResultString {
	parts := RequireTokens(text, 1, "Result")
	return ResultString{
		AddressMatches: parts[0],
	}
}

func (s ResultString) String() string {
	return Token(s.AddressMatches)
}

func (s ResultString) Equals(o ResultString) bool {
	return DNCEqual(s.AddressMatches, o.AddressMatches)
}

func EqualResultStringSlices(a, b []ResultString) bool {
	if len(a) != len(b) { return false }
	for i := range a {
		if !a[i].Equals(b[i]) { return false }
	}
	return true
}

File: Rust/ExampleTests/common/result_string.rs
#![allow(dead_code, unused_imports, unused_variables)]

#[derive(Debug, Clone, Default)]
pub struct ResultString {
    pub addressmatches: String,
}

impl ResultString {
    pub fn from_vec(v: &[&str]) -> Self {
        Self {
            addressmatches: v.get(0).copied().unwrap_or("").to_string(),
        }
    }

    /// Builds from the text form, e.g. Money as "25 USD".
    pub fn from_text(text: &str) -> Self {
        let parts = crate::common::tokens::require(text, 1, "Result");
        Self {
            addressmatches: parts[0].clone(),
        }
    }
}

impl std::fmt::Display for ResultString {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        write!(f, "{}", [crate::common::tokens::token(&self.addressmatches)].join(" "))
    }
}

impl PartialEq for ResultString {
    fn eq(&self, other: &Self) -> bool {
        crate::common::dnc_equal(&self.addressmatches, &other.addressmatches)
    }
}
